keanu-project/src/test/java/io/improbable/keanu/network/ProtobufTest.java

```diff
@@ -458,12 +475,17 @@ private void checkSaveableVertex(Class<? extends Vertex> vertexClass) {
         Map<String, Class> requiredParameters = new HashMap<>();
 
         for (Parameter parameter : parentConstructor.get(0).getParameters()) {
-            LoadVertexParam annotation = parameter.getAnnotation(LoadVertexParam.class);
-            assertThat("Annotation has to be present on all params for class: " + vertexClass, annotation,
-                is(notNullValue()));
-            assertThat("Annotation can only be used once for class: " + vertexClass, requiredParameters,
-                not(hasKey(annotation.value())));
-            requiredParameters.put(annotation.value(), parameter.getType());
+            LoadVertexParam parameterAnnotation = parameter.getAnnotation(LoadVertexParam.class);
+            LoadShape shapeAnnotation = parameter.getAnnotation(LoadShape.class);
+            assertThat("Annotation has to be present on all Constructor params for class: " + vertexClass,
+                parameterAnnotation != null || shapeAnnotation != null);
+            if (parameterAnnotation != null) {
+                assertThat("Annotation can only be used once for class: " + vertexClass, requiredParameters,
+                    not(hasKey(parameterAnnotation.value())));
+                requiredParameters.put(parameterAnnotation.value(), parameter.getType());
+            } else {
+                assertThat("Shape Arguments must be long[]", parameter.getType().isAssignableFrom(long[].class));
```

Comment:
The name of this method (`getRequiredParamsAndCheckOnlyUsedOnce`) doesn't make it that clear to me that anything is being done with the shape annotation. Is there a way we could either make that more clear or split it out from here?

Reply:
Renamed the function